Content.Server/Supermatter/EntitySystems/SupermatterSystem.cs

```diff
@@ -0,0 +1,426 @@
+
+                            factor = integrity switch
+                            {
+                                (<= 25) => 0.002f,
+                                (<= 55) and (> 25) => 0.005f,
+                                (<= 75) and (> 55) => 0.0009f,
+                                (<= 90) and (> 75) => 0.0005f,
+                                _ => 0,
+                            };
+
+                            damage = Math.Clamp((SMcomponent.Power * factor) * SupermatterComponent.DamageIncreaseMultiplier, 0f, SupermatterComponent.MaxSpaceExposureDamage);
+                        }
+                    }
+                }
+
+                //only take up to 1.8 damage per cycle with no lower limmit
+                damage = Math.Min(SMcomponent.DamageArchived + (SupermatterComponent.DamageHardcap * SupermatterComponent.ExplosionPoint), damage);
+                //damage to add to total
+                var damageDelta = new DamageSpecifier(_prototypeManager.Index<DamageTypePrototype>("Blunt"), (Shared.FixedPoint.FixedPoint2)damage);
+                _damageable.TryChangeDamage(SMcomponent.Owner, damageDelta, true);
+
+                SMcomponent.DamageUpdateAccumulator -= SupermatterComponent.DamageUpdateTimer;
+            }
+        }
+
+        /// </summary>
+        /// Runs the logic and timers for Delamination
+        /// </summary>
+        public void Delamination(EntityUid Uid, float frameTime, SupermatterComponent? SMcomponent = null, TransformComponent? Xform = null, ExplosiveComponent? Xplode = null)
+        {
+            if (!Resolve(Uid, ref SMcomponent, ref Xform, ref Xplode))
+            {
+                return;
+            }
+
+            var _delamType = SupermatterComponent.DelamType.Explosion;
+
+            var audioParams = AudioParams.Default;
+            audioParams.Loop = true;
+            audioParams.MaxDistance = 20f;
+            audioParams.Volume = 5;
+
+            //before we actually start counting down, check to see what delam type we're doing.
+            if (!SMcomponent.FinalCountdown)
+            {
+                //if we're in atmos
+                if (_atmosphere.GetTileMixture(Xform.Coordinates) is { } mixture)
+                {
+                    //if the moles on the sm's tile are above MolePenaltyThreshold
+                    if (mixture.TotalMoles >= SupermatterComponent.MolePenaltyThreshold)
+                    {
+                        _delamType = SupermatterComponent.DelamType.Singulo;
+                        _chatManager.DispatchStationAnnouncement(Loc.GetString("supermatter-delamination-overmass"), "Supermatter", false);
+                    }
+                }
+                else
+                {
+                    _delamType = SupermatterComponent.DelamType.Explosion;
+                    _chatManager.DispatchStationAnnouncement(Loc.GetString("supermatter-delamination-default"), "Supermatter", false);
+                }
+            }
```

Comment:
Is this from TG? Singulo delamination seems unspectacular compared to this:
https://ss13.moe/wiki/index.php/Supermatter_Cascade

Having the supermatter turn into a singulo should require something else than just pressure imo.

Reply:
this is a tg thing yeah